src/layman/rest_publication_test.py

```diff
@@ -46,3 +48,69 @@ def check_response(resp):
     r = patch_method(workspace, publication)
 
     delete_method(workspace, publication)
+
+
+class TestSoapClass:
+    username = 'test_rest_soap_user'
+    publ_name_prefix = 'test_rest_soap_'
+    authz_headers = process_client.get_authz_headers(username)
+    access_rights_rowo = {'read': f"{username}", 'write': f"{username}"}
+    access_rights_rewo = {'read': f"{username},EVERYONE", 'write': f"{username}"}
+    access_rights_rewe = {'read': f"{username},EVERYONE", 'write': f"{username},EVERYONE"}
+
+    @pytest.fixture(scope='class')
+    def reserve_username(self):
+        process_client.reserve_username(self.username, headers=self.authz_headers)
+        yield
+
+    @pytest.mark.usefixtures('liferay_mock', 'ensure_layman', 'reserve_username')
+    @pytest.mark.parametrize('params_and_expected_list', [
+        [(access_rights_rowo, False), (access_rights_rewe, True)],
```

Comment:
Zkusit, jak by to vypadalo s named tuples.

Reply:
I think not very well:
```
from collections import namedtuple

ParamsAndExpected = namedtuple('ParamsAndExpected', 'access_rights exp_public')

class TestSoapClass:

    ...
    @pytest.mark.parametrize('params_and_expected_list', [
        [ParamsAndExpected(access_rights_rowo, exp_public=False), ParamsAndExpected(access_rights_rewe, exp_public=True)],
        [ParamsAndExpected(access_rights_rewo, exp_public=True)],
        [ParamsAndExpected(access_rights_rewe, exp_public=True), ParamsAndExpected(access_rights_rowo, exp_public=False)],
    ])
    ...
    def test_soap_authz(self, publ_type, params_and_expected_list):
        ...
        for idx, params_and_expected in enumerate(params_and_expected_list):
            access_rights = params_and_expected.access_rights
            exp_public = params_and_expected.exp_public
        ...
```

Main outcome is `exp_public=`, but simple comment would probably make better job here. Everything else is just longer.

BTW ParamsAndExpected could be defined in TestSoapClass, but then you need to write `TestSoapClass.ParamsAndExpected`.